forLim.py

```diff
@@ -535,517 +548,552 @@ def run(self):
-                self.dlg.label_printActualProcess.setText("Add vector layer to map canevas ...")
-                QApplication.processEvents() 
-                
-                forest_path = os.path.join(args["Path_output"],"forest.shp")
-                if args["AddLayer"]:
-                    vlayer = QgsVectorLayer(forest_path, "forest", "ogr")
-                    QgsMapLayerRegistry.instance().addMapLayer(vlayer)
-                    
-                    forest = {
-                        'Foret dense':      (QColor(0,100,0), 'Foret dense'),
-                        'Paturage boise':   (QColor(0,160,0), 'Paturage boise'),
-#                        '' :                ('white', 'unknown')
-                    }
                     
-                    # create a category for each item in animals
-                    categories = []
-                    for forest_name, (color, label) in forest.items():
-                        symbol = QgsSymbolV2.defaultSymbol(vlayer.geometryType())
-                        symbol.setColor(QColor(color))
-                        category = QgsRendererCategoryV2(forest_name, symbol, label)
-                        categories.append(category)
-                    
-                    # create the renderer and assign it to a layer
-                    expression = "TYPE"
-                    renderer = QgsCategorizedSymbolRendererV2(expression, categories)
-                    vlayer.setRendererV2(renderer)
-                
-                
-                
-                #Print progress on user window
-                self.dlg.progressBar.setValue(self.dlg.progressBar.maximum())
-                self.dlg.label_printActualProcess.setText("Done.")
-                QApplication.processEvents() 
-    
-    
-####
-                
-                #Write metadata
-                infos_path = os.path.join(args["Path_output"], "metadata.txt")
-                file = open(infos_path, "w")
-                file.write("Processed: " + str(now_time.date()) + "\n\n###############\n\n")
-                for arg_i in args:
-                    file.write(arg_i + " : " + str(args[arg_i]) + "\n")
-                file.close()
-                
-                
+                    #Print progress on user window
+                    self.dlg.progressBar.setValue(self.dlg.progressBar.maximum())
+                    self.dlg.label_printActualProcess.setText("Done.")
+                    QApplication.processEvents()
+
+    ####
+                    
+                    #Write metadata
+                    infos_path = os.path.join(args["Path_output"], "metadata.txt")
+                    file = open(infos_path, "w")
+                    file.write("Processed: " + str(now_time.date()) + "\n\n###############\n\n")
+                    for arg_i in args:
+                        file.write(arg_i + " : " + str(args[arg_i]) + "\n")
+                    file.close()
+                    
+                    
```

Comment:
This is a quite big procedure in one shot. What about splitting it into distinct methods ?